app/models/locale_site_setting.rb

```diff
@@ -19,43 +22,35 @@ def self.language_names
     return @language_names if @language_names
 
     @lock.synchronize do
-      @language_names ||= YAML.load(File.read(File.join(Rails.root, 'config', 'locales', 'names.yml')))
+      @language_names ||= begin
+        names = YAML.load(File.read(File.join(Rails.root, 'config', 'locales', 'names.yml')))
+
+        DiscoursePluginRegistry.locales.each do |locale, options|
+          if !names.key?(locale) && options[:name] && options[:nativeName]
+            names[locale] = { "name" => options[:name], "nativeName" => options[:nativeName] }
```

Comment:
Is there a reason you're using string keys rather than symbols here?

Reply:
The `names` are loaded with `YAML.load` which also uses string keys. I didn't want to break anything, so I continued using string keys.